File: dsa-Python/Pattern/pattern12.java
import java.util.Scanner;
/**
 * pattern12
 */
public class pattern12 {

    public static void main(String[] args){
        Scanner sc =new Scanner(System.in);
        int limit=sc.nextInt();
        for (int i=1;i<=limit;i++){
            for (int j=1;j<=i;j++){
                System.out.print(j);
            }
            System.out.print(" ".repeat((limit-i)*2));
            for (int j=i;j>=1;j--){
                System.out.print(j);
            }
            System.out.println();
        }
    }
}

File: dsa-Python/Pattern/pattern12.py
limit=int(input())
for i in range(1,limit+1):
    n=1
    while (n<=i):
        print(n,end="")
        n+=1
    print(" "*((limit-i)*2),end="")
    n-=1
    while(n>=1):
        print(n,end="")
        n-=1
    print()